test/integration_test/applicationbackup_test.go

```diff
@@ -40,33 +47,40 @@ func triggerBackupRestoreTest(
 	logrus.Infof("App created %v. Starting backup.", ctx.GetID())
 
 	// Create backuplocation here programatically using config-map that contains name of secrets to be used, passed from the CLI
-	err = createBackupLocation(t, appKey+"-backup-location", ctx.GetID(), storkv1.BackupLocationS3, "secret-config")
-	require.NoError(t, err, "Error creating backuplocation")
+	if createBackupLocationFlag {
+		err = createBackupLocation(t, appKey+"-backup-location", ctx.GetID(), storkv1.BackupLocationS3, "secret-config")
+		require.NoError(t, err, "Error creating backuplocation")
+	}
 
 	// Backup application
 	err = schedulerDriver.AddTasks(ctxs[0],
 		scheduler.ScheduleOptions{AppKeys: appBackupKey})
 	require.NoError(t, err, "Error backing-up apps")
-	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
-	require.NoError(t, err, "Error waiting for back-up to complete.")
+	if backupSuccessExpected {
+		err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for back-up to complete.")
+		logrus.Infof("Backup completed. Starting Restore.")
 
-	logrus.Infof("Backup completed. Starting Restore.")
+		// Restore application
+		err = schedulerDriver.AddTasks(ctxs[0],
+			scheduler.ScheduleOptions{AppKeys: appRestoreKey})
+		require.NoError(t, err, "Error restoring apps")
 
-	// Restore application
-	err = schedulerDriver.AddTasks(ctxs[0],
-		scheduler.ScheduleOptions{AppKeys: appRestoreKey})
-	require.NoError(t, err, "Error restoring apps")
+		err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for restore to complete.")
 
-	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
-	require.NoError(t, err, "Error waiting for restore to complete.")
+		logrus.Infof("Restore completed.")
 
-	logrus.Infof("Restore completed.")
+		// Validate applications after restore
+		err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for restore to complete.")
 
-	// Validate applications after restore
-	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
-	require.NoError(t, err, "Error waiting for restore to complete.")
+		logrus.Infof("App validation after restore completed.")
 
-	logrus.Infof("App validation after restore completed.")
+	} else {
+		err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout/5, defaultWaitInterval)
+		require.Error(t, err, "Backup expected to fail with missing pre-exec rule.")
```

Comment:
Remove `with missing pre-exec rule` from the message since it could fail for other reasons too